redisson/src/main/java/org/redisson/connection/ReplicatedConnectionManager.java

```diff
@@ -192,33 +218,33 @@ private void checkNode(AsyncCountDownLatch latch, RedisURI uri, ReplicatedServer
                     }
                     RedisConnection connection = connectionFuture.toCompletableFuture().join();
                     InetSocketAddress addr = connection.getRedisClient().getAddr();
-                    Role role = Role.valueOf(r.get(ROLE_KEY));
+                    final Role role = Role.valueOf(r.get(ROLE_KEY));
                     if (Role.master.equals(role)) {
                         InetSocketAddress master = currentMaster.get();
                         if (master.equals(addr)) {
                             log.debug("Current master {} unchanged", master);
-                            return CompletableFuture.completedFuture(null);
                         } else if (currentMaster.compareAndSet(master, addr)) {
                             CompletableFuture<RedisClient> changeFuture = changeMaster(singleSlotRange.getStartSlot(), uri);
-                            return changeFuture.exceptionally(e -> {
+                            return changeFuture.handle((ignored, e) -> {
```

Comment:
handle != exceptionally and will be executed always

Reply:
Oh right, sorry I'll fix it

Reply:
done